scalding-core/src/main/scala/com/twitter/scalding/mathematics/Matrix2.scala

```diff
@@ -142,6 +142,19 @@ sealed trait Matrix2[R, C, V] extends Serializable {
     MatrixLiteral(result, SizeHint.asDiagonal(this.sizeHint.setRowsToCols)) * matD
   }
 
+  /**
+   * Row L1 normalization (can only be called for Double)
+   * After this operation, the sum(|x|) alone each row will be 1.
+   */
+  def rowL1Normalize(implicit ev: =:=[V, Double], mj: MatrixJoiner2): Matrix2[R, C, Double] = {
```

Comment:
can we do this:
```scala
def rowL1Normalize(implicit num: Numeric[V], mj: MatrixJoiner2): Matrix[R, C, Double] = {...
```
then use `num.toDouble(x).abs` on line 151? This will let us work with Int, Long, Float, etc...

Also if you could make the same change to rowL2Normalize, I'd be pumped!